### DISCUSSION TITLE
Comando FlutterTTs.speak dentro do body

### DISCUSSION THREAD

**Discussion by nambasoft** [Flutterando/forum#193] [Perguntas e Respostas]
Boa tarde, preciso executar o comando FlutterTTs.speak dentro do Widget buil, pego o valor de uma variavel. Nao quero usar nenhum botao, estilo em tempo real. Obrigado a Todos.

**Comment by eduardoflorence** [comment]
Melhor fornecer um trecho de código de exemplo para entendermos melhor sua dúvida

**Reply by nambasoft** [reply]
queria colocar nesta parte do codigo

Widget build(BuildContext context) {
  return Text("$text");
}


Aqui sai um texto e gostaria que o flutter_tts.speak("$texto);
falasse esse texto

**Reply by eduardoflorence** [reply]
Você pode criar um widget que mostra o texto e também fala. A parte da fala seria agendada após ele mostrar o texto.
Veja um exemplo:

```dart
class TextAndSpeek extends StatefulWidget {
  const TextAndSpeek(this.text, {super.key});

  final String text;

  @override
  State<TextAndSpeek> createState() => _TextAndSpeekState();
}

class _TextAndSpeekState extends State<TextAndSpeek> {
  @override
  void initState() {
    super.initState();
    // Aqui ele agenda a fala do texto para depois do desenho do Texto,
    // que está no build abaixo
    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
      flutter_tts.speak(widget.text);
    });
  }

  @override
  Widget build(BuildContext context) {
    return Text(widget.text);
  }
}
```
